Tools/YamlCreate.ps1

```diff
@@ -31,232 +31,663 @@ filter TrimString {
+    Write-Host ''
+    Write-Host -ForegroundColor 'Green' -Object '[Required] Enter the application name. For example: Teams'
+    $PackageName = Read-Host -Prompt 'Application Name' | TrimString
 }
 
-while ([string]::IsNullOrWhiteSpace($version)) {
-    $version = Read-Host -Prompt 'Enter the version. For example: 1.0.0, 1.0.0.0, 1.0' | TrimString
-    $ManifestName = $version + ".yaml"
+while ([string]::IsNullOrWhiteSpace($PackageVersion) -or $PackageName.Length -ge 128) {
+    Write-Host ''
+    Write-Host -ForegroundColor 'Green' -Object '[Required] Enter the version. for example: 1.33.7'
+    $PackageVersion = Read-Host -Prompt 'Version' | TrimString
 }
 
-while ([string]::IsNullOrWhiteSpace($License) -or $License.Length -ge 40) {
-    $License = Read-Host -Prompt 'Enter the License, For example: MIT, or Copyright (c) Microsoft Corporation' | TrimString
+$PackageIdentifier = [System.String]::Concat($Publisher.Replace(' ',''),'.',$PackageName.Replace(' ',''))
+if (Test-Path -Path "$PSScriptRoot\..\manifests") {
+    $ManifestsFolder = (Resolve-Path "$PSScriptRoot\..\manifests").Path
+} else {
+    $ManifestsFolder = (Resolve-Path ".\").Path
 }
 
-while ($InstallerType -notin @("exe", "msi", "msix", "inno", "nullsoft", "appx", "wix", "zip")) {
-    $InstallerType = Read-Host -Prompt 'Enter the InstallerType. For example: exe, msi, msix, inno, nullsoft'
-}
-
-while ($architecture -notin @("x86", "x64", "arm", "arm64", "neutral")) {
-    $architecture = Read-Host -Prompt 'Enter the architecture (x86, x64, arm, arm64, Neutral)'
-}
-
-do {
-    $LicenseUrl = Read-Host -Prompt '[OPTIONAL] Enter the license URL' | TrimString
-} while (-not [string]::IsNullOrWhiteSpace($LicenseUrl) -and ($LicenseUrl.Length -lt 10 -or $LicenseUrl.Length -gt 2000))
-
-do {
-    $AppMoniker = Read-Host -Prompt '[OPTIONAL] Enter the AppMoniker (friendly name/alias). For example: vscode' | TrimString
-} while ($AppMoniker.Length -gt 40)
-
-do {
-    $Tags = Read-Host -Prompt '[OPTIONAL] Enter any tags that would be useful to discover this tool. For example: zip, c++' | TrimString
-} while ($Tags.Length -gt 40)
-
-do {
-    $Homepage = Read-Host -Prompt '[OPTIONAL] Enter the Url to the homepage of the application' | TrimString
-} while (-not [string]::IsNullOrWhiteSpace($LicenseUrl) -and ($Homepage.Length -lt 10 -or $Homepage.Length -gt 2000))
-
-do {
-    $Description = Read-Host -Prompt '[OPTIONAL] Enter a description of the application' | TrimString
-} while ($Description.Length -gt 500)
-
-# Only prompt for silent switches if $InstallerType is "exe"
-if ($InstallerType -ieq "exe") {
-    $Silent = Read-Host -Prompt '[OPTIONAL] Enter the silent install switch'| TrimString
-    $SilentWithProgress = Read-Host -Prompt '[OPTIONAL] Enter the silent (with progress) install switch'| TrimString
+$AppFolder = Join-Path $ManifestsFolder $Publisher.Chars(0) $Publisher.Replace(' ','') $PackageName.Replace(' ','') $PackageVersion
```

Comment:
The path should always be the same as the packageId (split on '.'), so better to reverse engineer it from that. Additionally, PackageIds may have more than two tokens, for example https://github.com/microsoft/winget-pkgs/blob/master/manifests/m/Microsoft/VisualStudio/Community/16.9.31129.286/Microsoft.VisualStudio.Community.yaml